### PR TITLE
base architecture with `.docx` support

### PR DESCRIPTION
**Pull request by AHReccese**
#### Reference Issues/PRs
#1 #2 #3 #4 #5 #6 #7 #10 #14 #18 #22
#### What does this implement/fix? Explain your changes.
base architecture with `.docx` support + a huge refactoring in the template file.
* `clear` functionality
* `clear all` functionality
* `update` functionality
* `update all` functionality

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -62,13 +63,7 @@ def read_description():
...
         'Intended Audience :: Manufacturing',
         'Intended Audience :: Science/Research',
-        'Topic :: Scientific/Engineering :: Information Analysis',
-        'Topic :: Education',
-        'Topic :: Scientific/Engineering',
-        'Topic :: Scientific/Engineering :: Artificial Intelligence',
-        'Topic :: Scientific/Engineering :: Human Machine Interfaces',
-        'Topic :: Scientific/Engineering :: Mathematics',
-        'Topic :: Scientific/Engineering :: Physics',
+        'Topic :: Scientific/Security :: Metadata leakage',
```
Comment (sepandhaghighi) [inline setup.py:66]: This classifier is not a standard one. It is recommended that you choose from classifiers listed [here](https://pypi.org/classifiers/).

Comment (sepandhaghighi) [reply setup.py:66]: My suggestion:

1. `Topic :: Security`
2. `Topic :: Utilities`

File: dmeta/util.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    docx_file_name = remove_format(docx_file_name)
+    source_file = ZipFile(docx_file_name + ".docx")  
+    unzipped_dir = os.path.join(os.getcwd(), "unzipped_" + docx_file_name)
+    rmtree(unzipped_dir, ignore_errors=True)
+    os.mkdir(unzipped_dir)
+    source_file.extractall(unzipped_dir)
+    return unzipped_dir, source_file
+
+
+def read_json(config_file_name):
```
Comment (sepandhaghighi) [inline dmeta/util.py:60]: Docstring

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,13 +5,20 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
...
+- `remove_format` function in `util.py`
+- `extract_docx` function in `util.py`
+- `read_json` function in `util.py`
 
-[Unreleased]: https://github.com/openscilab/metadoc/compare/v0.1...dev
-[0.1]: https://github.com/openscilab/metadoc/compare/3598e8b...v0.1
\ No newline at end of file
+
+[Unreleased]: https://github.com/openscilab/dmeta/compare/v0.1...dev
+[0.1]: https://github.com/openscilab/dmeta/compare/3598e8b...v0.1
```
Comment (sepandhaghighi) [inline CHANGELOG.md:24]: `9a4ad10` instead of `3598e8b`

**Review by sepandhaghighi — CHANGES_REQUESTED** [review]
@AHReccese Thanks for your efforts 🔥 
Please take a look at my comments 🥇

File: setup.py (hunk 1 above)
Comment (AHReccese) [reply setup.py]: done.

File: dmeta/util.py (hunk 2 above)
Comment (AHReccese) [reply dmeta/util.py:60]: done.

File: CHANGELOG.md (hunk 3 above)
Comment (AHReccese) [reply CHANGELOG.md]: done.

**Comment by AHReccese** [comment]
All of the requested changes have been applied accordingly. 
Let me know if you have any other.
@sepandhaghighi

File: setup.py [hunk 4]
```diff
@@ -23,24 +23,25 @@ def read_description():
...
+        'dmeta',
+        'dmeta.utils'],
     version='0.1',
-    description='Transportation of ML models',
+    description='Removing microsoft office files\' metadata',
     long_description=read_description(),
     long_description_content_type='text/markdown',
-    author='MetaDoc Development Team',
+    author='dmeta Development Team',
     author_email='info@pycm.io',
```
Comment (sepandhaghighi) [inline setup.py:39]: dmeta@openscilab.com

File: setup.py [hunk 5]
```diff
@@ -23,24 +23,25 @@ def read_description():
...
-        'metadoc',],
+        'dmeta',
+        'dmeta.utils'],
     version='0.1',
-    description='Transportation of ML models',
+    description='Removing microsoft office files\' metadata',
     long_description=read_description(),
     long_description_content_type='text/markdown',
-    author='MetaDoc Development Team',
+    author='dmeta Development Team',
```
Comment (sepandhaghighi) [inline setup.py:38]: DMeta

File: setup.py [hunk 6]
```diff
@@ -23,24 +23,25 @@ def read_description():
             description += c.read()
         return description
     except Exception:
-        return '''Transportation of ML models'''
+        return '''Removing metadata'''
 
 
 setup(
-    name='metadoc',
+    name='dmeta',
```
Comment (sepandhaghighi) [inline setup.py:30]: DMeta

File: dmeta/functions.py [hunk 7]
```diff
@@ -0,0 +1,184 @@
+# -*- coding: utf-8 -*-
+"""dmeta Functions."""
```
Comment (sepandhaghighi) [inline dmeta/functions.py:2]: `functions` instead of `Functions`

File: dmeta/params.py [hunk 8]
```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+"""dmeta Parameters and constants."""
```
Comment (sepandhaghighi) [inline dmeta/params.py:2]: `parameters` instead of `Parameters`

**Review by sepandhaghighi — CHANGES_REQUESTED** [review]
@AHReccese Thanks for your efforts 💯 
I suggest using `DMeta` instead of `dmeta` in titles (I mentioned some of them in the code).

**Comment by AHReccese** [comment]
I tried to update `dmeta` to `DMeta` in every place I found. please let me know if there are any others.

**Comment by AHReccese** [comment]
@sepandhaghighi 
Dear Sepand please forget the CI/CD failure in this PR and finalize your review.
I will push up a separate Test & CI/CD dedicated PR after that.

**Review by sepandhaghighi — APPROVED** [review]
@AHReccese 

I believe it would be better to use the `defusedxml` package for XML parsing. The default XML package has lots of known and unknown vulnerabilities.
As this change will result in significant changes, I suggest handling it in a separate PR.

**Pull request merged**